### PR TITLE
Bring bwa/bwa-mem up to IUC standards

### PR DESCRIPTION
**Pull request by mvdbeek**
- 4 space indentation consistently
- Command and tokens without leading indentation
- Double-quotes replaced with single quotes

Also adds multihreaded samtools sorting, which should really
speedup bwa with high GALAXY_SLOTS variables.

If you are going to review this it is probably a good idea to look
at the individual commits, to not be thrown off by the indentation
changes.

### CODE REVIEW

File: tools/bwa/bwa_macros.xml [hunk 1]
```diff
@@ -1,115 +1,115 @@
...
-              </when>
-          </conditional>
-      </actions>
-  </macro>
-
-  <token name="@RG@">
+    <xml name="requirements">
+        <requirements>
+            <requirement type="package" version="@VERSION@">bwa</requirement>
+            <requirement type="package" version="1.6">samtools</requirement>
```
Comment (nsoranzo) [inline tools/bwa/bwa_macros.xml]: You need to increase the tools' version or revert this change.

File: tools/bwa/bwa.xml [hunk 2]
```diff
@@ -1,457 +1,466 @@
...
+                       label="Select first set of reads" help="Specify dataset with forward reads"/>
+                <param name="fastq_input2" type="data" format="fastqsanger,fastqsanger.gz,fasta"
+                       label="Select second set of reads" help="Specify dataset with reverse reads"/>
+                <conditional name="adv_pe_options">
+                    <expand macro="advanced_pe_options"/>
+                </conditional>
+            </when>
+            <when value="paired_collection">
+                <param name="fastq_input1" format="fastqsanger,fastqsanger.gz" type="data_collection"
+                       collection_type="paired" label="Select a paired collection"
```
Comment (nsoranzo) [inline tools/bwa/bwa.xml]: Can you put `collection_type="paired"` on the previous line?

File: tools/bwa/bwa.xml [hunk 3]
```diff
@@ -1,457 +1,466 @@
...
+options:
+
+1. **Use a built-in genome index** - when selected (this is default), Galaxy provides the user with **Select
+reference genome index** dropdown. Genomes listed in this dropdown have been pre-indexed with bwa index utility
+and are ready to be mapped against.
+2. **Use a genome from the history and build index** - when selected, Galaxy provides the user with **Select
+reference genome sequence** dropdown. This dropdown is populated by all FASTA formatted files listed in your
+current history. If your genome of interest is uploaded into history it will be shown there. Selecting a genome
+from this dropdown will cause Galaxy to first transparently index it using `bwa index` command, and then run
+mapping with `bwa aln`.
```
Comment (nsoranzo) [inline tools/bwa/bwa.xml]: Need to indent these lines to have an enumerated list, see http://docutils.sourceforge.net/docs/user/rst/quickref.html#enumerated-lists

File: tools/bwa/bwa.xml [hunk 4]
```diff
@@ -1,457 +1,466 @@
...
-  2. **Use a genome from the history and build index** - when selected, Galaxy provides the user with **Select reference genome sequence** dropdown. This dropdown is populated by all FASTA formatted files listed in your current history. If your genome of interest is uploaded into history it will be shown there. Selecting a genome from this dropdown will cause Galaxy to first transparently index it using `bwa index` command, and then run mapping with `bwa aln`.
-    
 If your genome of interest is not listed here you have two choices:
 
-  1. Contact galaxy team using **Help->Support** link at the top of the interface and let us know that an index needs to be added
-  2. Upload your genome of interest as a FASTA file to Galaxy history and selected **Use a genome from the history and build index** option.
+1. Contact galaxy team using **Help->Support** link at the top of the interface and let us know that an index
+needs to be added
+2. Upload your genome of interest as a FASTA file to Galaxy history and selected **Use a genome from the history
+and build index** option.
```
Comment (nsoranzo) [inline tools/bwa/bwa.xml]: Need to indent these lines to have an enumerated list, see http://docutils.sourceforge.net/docs/user/rst/quickref.html#enumerated-lists

File: tools/bwa/bwa.xml [hunk 5]
```diff
@@ -1,457 +1,466 @@
...
 
-Galaxy allows three levels of control over bwa-mem options provided by **Select analysis mode** menu option. These are:
+Galaxy allows three levels of control over bwa-mem options provided by **Select analysis mode** menu option.
+These are:
 
-  1. *Simple Illumina mode*: The simplest possible bwa mem application in which it alignes single or paired-end data to reference using default parameters. It is equivalent to the following command: bwa mem &lt;reference index&gt; &lt;fastq dataset1&gt; [fastq dataset2]
-  2. *Full list of options*: Allows access to all options through Galaxy interface.
+1. *Simple Illumina mode*: The simplest possible bwa mem application in which it alignes single or paired-end
+data to reference using default parameters. It is equivalent to the following command: bwa mem &lt;reference
+index&gt; &lt;fastq dataset1&gt; [fastq dataset2]
```
Comment (nsoranzo) [inline tools/bwa/bwa.xml]: Need to indent these lines to have an enumerated list, see http://docutils.sourceforge.net/docs/user/rst/quickref.html#enumerated-lists

File: tools/bwa/bwa.xml [hunk 6]
```diff
@@ -1,457 +1,466 @@
...
-    -q INT    quality threshold for read trimming down to 35bp [0]
-    -B INT    length of barcode
-    -L        log-scaled gap penalty for long deletions
-    -N        non-iterative mode: search for all n-difference hits (slooow)
-    -I        the input is in the Illumina 1.3+ FASTQ-like format
-    -b        the input read file is in the BAM format
-    -0        use single-end reads only (effective with -b)
-    -1        use the 1st read in a pair (effective with -b)
-    -2        use the 2nd read in a pair (effective with -b)
+-n NUM max #diff (int) or missing prob under 0.02 err rate (float) [0.04]
```
Comment (nsoranzo) [inline tools/bwa/bwa.xml]: I'd get rid of all the option descriptions below.

File: tools/bwa/bwa.xml (hunk 5 above)
Comment (nsoranzo) [reply tools/bwa/bwa.xml]: Also, the various `&gt;` and `&lt;` need to be replaced with `>` and `<` respectively, now that they are inside CDATA.

Comment (mvdbeek) [reply tools/bwa/bwa.xml]: These options only exist in bwa-mem, I'll just remove this.

**Comment by nsoranzo** [comment]
Thanks @mvdbeek!
